pkg/cloud/aws/converters/instance.go

```diff
@@ -36,14 +38,12 @@ func SDKToInstance(v *ec2.Instance) *v1alpha1.Instance {
 		EBSOptimized: v.EbsOptimized,
 	}
 
+	i.IAMProfile = strings.Split(aws.StringValue(v.IamInstanceProfile.Arn), "instance-profile/")[1]
```

Comment:
We should make sure that the value isn't nil or empty string and check if the length of the slice returned from Split is actually greater or equal to 1. The line above can cause a panic otherwise.

Reply:
Updated.